src/Animal.java

```diff
@@ -0,0 +1,24 @@
+public abstract class Animal {
+
+        private static int count = 0;
+        private int maxRun ;
```

Comment:
where is the field value assigned?

add assignment code into constructor
`this.maxRun = maxRun;`